src/verify_cert.rs

```diff
@@ -1096,14 +1064,57 @@ mod tests {
 
         verify_chain(
             anchors,
-            &intermediates_der,
+            &intermediate_chain.chain,
             &ee_cert,
             Some(&expected_chain),
             None,
         )
         .map(|_| ())
     }
 
+    fn build_linear_chain(
+        ca_cert: &CertifiedKey,
+        chain_length: usize,
+        all_same_subject: bool,
+    ) -> IntermediateChain {
+        let mut intermediates: Vec<CertifiedKey> = Vec::with_capacity(chain_length);
+
+        for i in 0..chain_length {
+            let (issuer, issuer_key) = if i == 0 {
+                (&ca_cert.cert, &ca_cert.key_pair)
+            } else {
+                (&intermediates[i - 1].cert, &intermediates[i - 1].key_pair)
+            };
+            let intermediate = issuer_params(match all_same_subject {
+                true => "Bogus Subject".to_string(),
+                false => format!("Bogus Subject {i}"),
+            });
+            let intermediate_key_pair =
+                KeyPair::generate_for(test_utils::RCGEN_SIGNATURE_ALG).unwrap();
```

Comment:
I guess you missed this one because it's new? :)

Reply:
Hm? Not sure I follow. It's now consistent with the other places we generate a keypair. I removed the helper and switched users of it, or `KeyPair::generate()` to use the explicit `KeyPair::generate_for` with `RCGEN_SIGNATURE_ALG`.

Reply:
Okay, I misread your other comment -- never mind.